### ISSUE TITLE
How can one tell if an evar is named?

### ISSUE THREAD

**Issue by cpitclaudel** [coq/coq#11084]
#### Description of the problem

This is likely just a documentation request, though there may be a possible UI improvement here too.  Consider the following:

```coq
Goal 0 = 0.
  etransitivity.
(*  ============================
    0 = ?y *)
  Fail [y]: exact 0. (* No such goal. *)
  instantiate (1 := ?[y]).
(*  ============================
    0 = ?y *)
  [y]: exact 0.
(*  ============================
    0 = 0 *)
```

The first two goals look exactly the same.  How can one tell whether the name `?y` can be referred to or not using a goal selector?

(I realize that it's likely better not to rely on auto-generated names, so it's not necessarily an issue that the first name can't be referred to; what's puzzling is that there doesn't seem to be a visual indication that distinguishes a name that *can* be referred to from one that *cannot* be referred to.)

Maybe I missed a setting in the docs?  Alternatively, maybe that second `?y` should be printed as `?[y]`? Or maybe both should be addressable as `?y`

Thanks!

#### Coq Version

8.9.1

**Comment by Zimmi48** [comment]
I completely agree. @herbelin What should we do about this? Follow the SSReflect convention for non-addressable names?

**Comment by herbelin** [comment]
No strong opinion. We could use an underscore (i.e. `_y`) for the first case. Or we could allow `[y]: exact 0`. One debated issue about the second case was the lack of robustness in relying on a generated name. But we could also accept it with a warning.

**Comment by Zimmi48** [comment]
> We could use an underscore

Does it correspond to SSR's convention, or is it a tilde `~`?

> we could also accept it with a warning

That would make sense too.

**Comment by JasonGross** [comment]
Wasn't there a performance issue with accepting named references to automatically named evars?

**Comment by herbelin** [comment]
@jasongross: The performance issue was, iirc, to systematically generate a fresh name for internal, and moreover non-dependent, goals, hence not needed for printing (the typical example being a `destruct t` with thousands of subgoals).
